core/isolate.rs

```diff
@@ -688,11 +638,18 @@ impl Future for Isolate {
 
   fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
     let inner = self.get_mut();
-
     inner.waker.register(cx.waker());
-
     inner.shared_init();
 
+    let mut locker = v8::Locker::new(&*inner.v8_isolate.as_mut().unwrap());
+    let mut hs = v8::HandleScope::new(locker.enter());
+    let scope = hs.enter();
+    let context = inner.global_context.get(scope).unwrap();
+    let mut cs = v8::ContextScope::new(scope, context);
+    let scope = cs.enter();
+
+    inner.check_promise_exceptions(scope)?;
```

Comment:
Seems suspicious, why is it needed here @piscisaureus ?

Reply:
Not strictly needed. But what could happen is that there are multiple unhandled-exception Promises around. This makes sure that we eventually flush them out.

The old code would loop over the entire list of promise exceptions and discard all but the last one. This seemed better.

You can test it yourself:

### v0.30.0

```
D:\deno2>deno
> async function x() { throw new Error("BUH"); }
undefined
> x(), x()
Promise {}
> error: Uncaught Error: BUH
► <unknown>:1:28
    at x (<unknown>:1:28)
    at <unknown>:1:1
    at evaluate ($deno$/repl.ts:84:34)
    at replLoop ($deno$/repl.ts:175:13)
```

### This PR:

```
D:\deno2>target\debug\deno
> async function x() { throw new Error("BEEH"); }
undefined
> x(), x()
Promise {}
error: Uncaught Error: BEEH
► <unknown>:1:28
    at x (<unknown>:1:28)
    at <unknown>:1:1
    at evaluate ($deno$/repl.ts:83:34)
    at replLoop ($deno$/repl.ts:174:13)
error: Uncaught Error: BEEH
► <unknown>:1:28
    at x (<unknown>:1:28)
    at <unknown>:1:6
    at evaluate ($deno$/repl.ts:83:34)
    at replLoop ($deno$/repl.ts:174:13)
>   
```

Reply:
Awesome 👍

Reply:
Consider adding this as a test case (preferably a unit test in core rather than integration test in Cli)

Reply:
> Consider adding this as a test case (preferably a unit test in core rather than integration test in Cli)

Yeah I'm actually trying but in core it doesn't work so well. 
I'm also discovering some other (pre-existing) issues.
So I'll land this and add some responsible-adult in a bit.